tests/testament/tester.nim

```diff
@@ -404,6 +404,21 @@ proc testC(r: var TResults, test: TTest) =
     if exitCode != 0: given.err = reExitCodesDiffer
   if given.err == reSuccess: inc(r.passed)
 
+proc testExec(r: var TResults, test: TTest) =
```

Comment:
It is still a bit dubious why this is required, but it's much better than before. Will merge once the tests are green.